syft/rekor/rekor_test.go

```diff
@@ -0,0 +1,223 @@
+	"github.com/go-openapi/runtime"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/client/entries"
+	"github.com/sigstore/rekor/pkg/generated/client/index"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spdx/tools-golang/spdx"
+	"github.com/stretchr/testify/assert"
+)
+
+type testCase struct {
+	name                string
+	inputFilePath       string // resolver and location are created from this
+	expectedOutput      []artifact.Relationship
+	uuidsToLogEntryFile map[string]string // tells the rekorClient mock what uuids and what log entries to return
+	httpClient          *http.Client
+	expectedErr         string
+	expectedLog         string
+}
+
+/*
+***************
+
+	Functions and types to complete the interfaces in client.Rekor. Most are unimplemented.
+
+****************
+*/
+
+type entriesMock struct {
+	test testCase
+}
+
+type indexMock struct {
+	test testCase
+}
+
+type roundTripperMock struct {
+	sbomFile string
+}
+
+func (rt roundTripperMock) RoundTrip(req *http.Request) (*http.Response, error) {
+	sbom_bytes, err := os.ReadFile(rt.sbomFile)
+	if err != nil {
+		return nil, err
+	}
+	return &http.Response{
+		StatusCode: 200,
+		Body:       io.NopCloser(bytes.NewReader(sbom_bytes)),
+	}, nil
+}
+
+func (e entriesMock) CreateLogEntry(params *entries.CreateLogEntryParams, opts ...entries.ClientOption) (*entries.CreateLogEntryCreated, error) {
+	return nil, errors.New("Unimplemented")
+}
+
+func (e entriesMock) GetLogEntryByIndex(params *entries.GetLogEntryByIndexParams, opts ...entries.ClientOption) (*entries.GetLogEntryByIndexOK, error) {
+	return nil, errors.New("Unimplemented")
+}
+
+func (e entriesMock) SearchLogQuery(params *entries.SearchLogQueryParams, opts ...entries.ClientOption) (*entries.SearchLogQueryOK, error) {
+	return nil, errors.New("Unimplemented")
+}
+
+func (e entriesMock) SetTransport(transport runtime.ClientTransport) {}
+
+func (a indexMock) SetTransport(transport runtime.ClientTransport) {}
+
+func (e entriesMock) GetLogEntryByUUID(params *entries.GetLogEntryByUUIDParams, opts ...entries.ClientOption) (*entries.GetLogEntryByUUIDOK, error) {
+	uuid := params.EntryUUID
+
+	filePath, ok := e.test.uuidsToLogEntryFile[uuid]
+	if !ok {
+		return nil, errors.New("no test data file exists for uuid %v")
+	}
+
+	if filePath == "return nil payload" {
+		return &entries.GetLogEntryByUUIDOK{
+			Payload: nil,
+		}, nil
+	}
+
+	entry := &models.LogEntry{}
+	bytes, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading test data from file %v: \n%w", filePath, err)
+	}
+	err = json.Unmarshal(bytes, entry)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling json test data from file %v: \n%w", filePath, err)
+	}
+
+	res := &entries.GetLogEntryByUUIDOK{
+		Payload: *entry,
+	}
+
+	return res, nil
+}
+
+// ignores input hash; just returns uuids in test case
```

Comment:
nit: prefix with function name so it will appear in godocs `// SearchIndex ignores...`